VideoPress: Use playback tokens to get resources when the VideoPress site privacy is set to private

lhkowalski opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

Fixes #27314, #26938 and #27391.

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->
* Renamed the `videoIsPrivate` flag to `needsPlaybackToken`
* Changed the resolver to rely on the new `needsPlaybackToken` flag when deciding if a video needs a playback token
* Exposed the `needsPlaybackToken` flag for all the videos on the client initial state

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

N/A.

#### Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

No.

#### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Check the "Video Privacy: Restrict views to members of this site" setting:

<img width="408" alt="Screen Shot 2022-11-11 at 19 16 20" src="https://user-images.githubusercontent.com/6760046/201438389-1d52d728-9440-4cbe-ad57-37b702ae7d36.png">

* Upload some videos to the media library, leaving the privacy setting as default
* Open the browser console to inspect errors and the network activity
* Do a hard refresh on the page
* Before the fix, the thumbnails for the added videos were broken:

<img width="1140" alt="Screen Shot 2022-11-11 at 19 32 20" src="https://user-images.githubusercontent.com/6760046/201439907-3f41fcda-1f29-45d8-a97b-4516041ed3d6.png">

* After the fix, you should see all the thumbnais properly:

<img width="1112" alt="Screen Shot 2022-11-11 at 19 22 52" src="https://user-images.githubusercontent.com/6760046/201439104-61a68292-8815-47d2-bf3a-8311657f525b.png">

* Before the fix, you would see failed `403` responses for the thumbnail files:

<img width="755" alt="Screen Shot 2022-11-11 at 19 33 18" src="https://user-images.githubusercontent.com/6760046/201439925-91edfda2-acda-47fc-819d-3756076cdbd1.png">

* After the fix, no `403` responses should appear on the console
* After the fix, you also will see requests to the playback token endpoint on the network log, showing that the application is now requesting the tokens that it will need:

<img width="597" alt="Screen Shot 2022-11-11 at 19 26 15" src="https://user-images.githubusercontent.com/6760046/201439392-f3b8faa2-8e67-4181-ad16-7d90decdb9b4.png">

* Make sure to test the list view as well as the grid view

lhkowalski committed:
Provide videoIsPrivate flag on the client initial state

Also, use the flag as the reference to decide if the poster should be null on the first load, preventing the load of resources that will 403 while we don't fetch the required playback token.

lhkowalski committed:
Add support to the new video_is_private response field on the VideoPress video types

The field is called videoIsPrivate on the types.

lhkowalski committed:
Map the new field from the response object to the video object

lhkowalski committed:
Use the new videoIsPrivate flag to decide if the video needs a playback token

lhkowalski committed:
Change list component to use the new videoIsPrivate as the source of the privacy state

lhkowalski committed:
Return list component privacy source to original value so it reflects the actual setting of the video

This way, despite the video being private when the site is private as well, the user see the actual setting on the video list.

lhkowalski committed:
Add changelog file

retrofox commented on projects/packages/videopress/src/client/state/resolvers.js:
```diff
@@ -18,8 +18,6 @@ import {
 	SET_LOCAL_VIDEOS_QUERY,
 	WP_REST_API_USERS_ENDPOINT,
 	WP_REST_API_VIDEOPRESS_PLAYBACK_TOKEN_ENDPOINT,
-	VIDEO_PRIVACY_LEVELS,
-	VIDEO_PRIVACY_LEVEL_PRIVATE,
```
should we remove these constants, too?

lhkowalski replied:
@retrofox Remove the import? Yes, because they are not being used in any other place of the file.

lhkowalski committed:
Rename the videoIsPrivate flag to a more contextual needsPlaybackToken

This way, it's value gets tied to the idea of the playback token need, that is the usage that we were making for the flag anyway.

dhasilva reviewed (approved):
LGTM

lhkowalski committed:
Merge branch 'trunk' into fix/use-playback-token-when-site-is-private
